### ISSUE TITLE
Problems with PostgreSQL Storage

### ISSUE THREAD

**Issue by dandago** [dotnet/orleans#3085]
In relation to #3070, I am checking out the ADO .NET Storage Provider with PostgreSQL. The [Grain Persistence](https://dotnet.github.io/orleans/Documentation/Getting-Started-With-Orleans/Grain-Persistence.html) page does suggest that PostgreSQL is supported, and the Microsoft.Orleans.OrleansSqlUtils package does have a script for PostgreSQL. However:

1. There is no storage table in the script for PostgreSQL:

![image](https://cloud.githubusercontent.com/assets/13241564/26753669/9748c1c6-486b-11e7-834a-d9ee66c1c9c3.png)

2. There is no mention anywhere of what the AdoInvariant for PostgreSQL should be, not even in the Relational Storage page where the ones for SQL Server and MySQL are listed. It's "Npgsql", and needs to be added to the documentation.

3. I was trying to set it up anyway, on a basic Dev/Test Host, using the following configuration:

```
    siloHost.Config.Globals.RegisterStorageProvider("Orleans.Storage.AdoNetStorageProvider",
        "OrleansStorage", new Dictionary<string, string>()
        {
            { "DataConnectionString", "Server=localhost;database=orleansstorage;uid=xxx;pwd=xxx;" },
            { "UseJsonFormat", "true" },
            { "AdoInvariant", "Npgsql" }
        });
```

The silo simply hangs after registering storage providers, and does not continue:

![image](https://cloud.githubusercontent.com/assets/13241564/26753685/1b2b7632-486c-11e7-9dbb-467e941fc597.png)

Repro project: 
[postgresql-orleans-repro.zip](https://github.com/dotnet/orleans/files/1049604/postgresql-orleans-repro.zip)

**Comment by dandago** [comment]
I see now that this has already been reported (#2991). PostgreSQL persistence is simply not implemented. Thus, the course of action I suggest is to:

* Remove reference to PostgreSQL from the persistence documentation until it is fully supported.
* Possibly investigate why the application hangs. It shouldn't hang without so much as an exception, period.

**Comment by sergeybykov** [comment]
The documentation part was addressed in #3100.
The application hangs are believed to be caused by https://github.com/npgsql/npgsql/issues/1130, and should have been mitigated by #3164.

I think we can close this one.
